great_expectations/expectations/core/expect_table_row_count_to_equal_other_table.py

```diff
@@ -296,23 +296,29 @@ def get_validation_dependencies(
         kwargs = configuration.kwargs if configuration else {}
         other_table_name = kwargs.get("other_table_name")
 
-        # create copy of table.row_count metric and modify "table" metric domain kwarg to be other table name  # noqa: E501 # FIXME CoP
-        table_row_count_metric_config_other: Optional[MetricConfiguration] = deepcopy(
+        # At this time, this is the only Expectation that
+        # computes the same metric over more than one domain
+        # ValidationDependencies does not allow duplicate metric names
+        # and the registry is checked to ensure the metric name is registered
```

Comment:
looks like you got this to work, so not blocking. But i am curious, where does this check happen? i looked for it, and it wasn't obvious. Seems like a more robust solution would solve this outside of the expectation.

Reply:
I agree that we could solve this in another way, but it represents a pretty big change in the metrics system that I don't think we want to take on in this PR.

Reply:
There is a super `BatchExpectation.get_validation_dependencies()` call that calls [this function](https://github.com/great-expectations/great_expectations/blob/710167a370d2d9932e19c133af265b3d20e08b1b/great_expectations/expectations/registry.py#L345), which checks if the metric exists in the registry. I will add that to the comment.